src/Timer.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { TimerProps, TimerState } from './models/models';
+
+const START_TIME = 100;
+
+class Timer extends React.Component<TimerProps, TimerState> {
+
+  state = { time: { minutes: 1, seconds: 0 }, songCount: 1};
+
+  constructor(props: TimerProps) {
+    super(props);
+    this.startTimer = this.startTimer.bind(this);
+  }
+
+  componentDidMount() {
+    this.startTimer();
+  }
+
+  async startTimer() {
+    // Loop through a list of [0,1,2,3...START_TIME-1]
+    for (const num of Array(START_TIME).keys()) {
+      // Set a tick to occur every 1.000 * n seconds
+      await setTimeout(this.tick, 1000*num, num);
+    }
+  }
+
+  tick = async (num: number) => {
+    // Unroll this.state.time into new object (basically clone it)
+    let newTime = { ...this.state.time };
```

Comment:
As a general rule in react, you never really want to modify the state objects directly. (I'm pretty sure) As convention you should stick to cloning the element you want to modify, modifying it, and then using `setState`. 

Here I'm using the spread operator `...`, see https://basarat.gitbook.io/typescript/future-javascript/spread-operator

Reply:
Ah gotcha, I am familiar with the spread operator syntax, although I didn't realize it could be applied to objects and not just arrays

Reply:
Yeah super useful for basically cloning objects or making a new object based off of another one that you just want to add a few new fields to